File: Python/Bubble_Sort.py
# Bubble Sort -
# fuction to swap elements -
def swapValue(firstElement, adjacentElement, arrToSort):
    (arrToSort[firstElement], arrToSort[adjacentElement]) = (arrToSort[adjacentElement], arrToSort[firstElement])
    
    # return value
    return arrToSort

# function to bubble sort list -
def bubbleSort(arrToSort):

    # No Element Swaped Indicator - 
    swaped = False
    
    # 2 loops with till lenght of list -
    # loop one carries firstElement
    for firstElement in range(0, len(arrToSort)):
    # loop two carries adjacentElement (Next Element in List)
        # this loop runs and reduces its lenght every time -
        for adjacentElement in range(0, len(arrToSort)-firstElement-1):
        # if firstElement value > adjacentElement 
            if arrToSort[adjacentElement] > arrToSort[adjacentElement+1]:
                # swap those digits and save to list -
                swapValue(adjacentElement, adjacentElement+1, arrToSort)
                swaped = True

        # if not element's swaped means array is already sorted
        if swaped == False:
            break


    # return sorted list
    return arrToSort

# Take Data In -
arrToSort = list(map(int, input().split()))[:]

# print(len(arrToSort))
print('Unsorted List = ', arrToSort, '\nSorted List =' ,bubbleSort(arrToSort))

File: JavaScript/Bubble_Sort.js
// Bubble Sort - 

// Swaping Function -
function swap(){
    var Temp = arrToSort[First_Element];
    arrToSort[First_Element] = arrToSort[Adjacent_Element];
    arrToSort[Adjacent_Element] = arrToSort[First_Element];
}

// Bubble Sort Function -
function bubbleSort(arrToSort){
    // 2 loops -
    // First loop takes First_Element
    for (let First_Element = 0; First_Element < arrToSort.lenght; First_Element++){
        // Second loop takes Adjacent_Element (i.e Next Element) -
        for(let Adjacent_Element = 0; Adjacent_Element < arrToSort.lenght; Adjacent_Element++){
            // Check If First_Element is < than Adjacent_Element 
            // If Yes than swap those values -
            swap(arrToSort[First_Element], arrToSort[Adjacent_Element], arrToSort)
        }
    } 
    // return Sorted Array -
    return arrToSort
}


// Take Array To Sort -
var arrToSort = [2, 5, 0, 1, 7, 5];

// Show User Sorted Array -
console.log(bubbleSort(arrToSort));
